cfme/common/provider.py

```diff
@@ -101,6 +103,46 @@ def deployment_helper(self, deploy_args):
         """ Used in utils.virtual_machines and usually overidden"""
         return {}
 
+    @property
+    def default_endpoint(self):
+        try:
+            return self.endpoints['default']
+        except (KeyError, AttributeError):
+            return None
+
+    @property
+    def hostname(self):
+        if self.default_endpoint:
+            return self.default_endpoint.hostname
+        else:
+            return None
+
+    @hostname.setter
+    def hostname(self, value):
+        if self.default_endpoint:
+            if value:
+                self.default_endpoint.hostname = value
+        else:
+            logger.warn("can't set hostname because default endpoint is absent")
+
+    @property
+    def ip_address(self):
+        if hasattr(self.default_endpoint, 'ipaddress'):
+            return self.default_endpoint.ipaddress
+        else:
+            if self.hostname:
+                return resolve_hostname(self.hostname)
+            else:
+                return None
+
+    @ip_address.setter
+    def ip_address(self, value):
+        if self.default_endpoint:
+            if value:
+                self.default_endpoint.ipaddress = value
+        else:
+            logger.warn("can't set ipaddress because default endpoint is absent")
```

Comment:
**OPTIONAL**
```python
try:
    if value:
        setattr(self.default_endpoint, "ipaddress", value)
except AttributeError:
    logger.warn("can't set ipaddress because default endpoint is absent")
```